packages/core/src/loaders/imageLoader.ts

```diff
@@ -183,6 +197,116 @@ export function loadAndCacheImages(
+ * The image is stored in the cache.
+ *
+ * @param imageId -  A Cornerstone Image Object's imageId
+ * @param options - Options to be passed to the Image Loader
+ *
+ * @returns Image Loader Object
+ */
+export function createAndCacheDerivedImage(
+  imageId: string,
+  options: DerivedImageOptions
+): Promise<IImage> {
+  if (imageId === undefined) {
+    throw new Error(
+      'createAndCacheDerivedImage: parameter imageId must not be undefined'
+    );
+  }
+  const imagePlaneModule = metaData.get('imagePlaneModule', imageId);
+
+  const length = imagePlaneModule.rows * imagePlaneModule.columns;
+  let numBytes, TypedArray;
+
+  const { targetBuffer } = options;
+  if (targetBuffer) {
+    if (targetBuffer.type === 'Float32Array') {
+      numBytes = length * 4;
+      TypedArray = Float32Array;
+    } else if (targetBuffer.type === 'Uint8Array') {
+      numBytes = length;
+      TypedArray = Uint8Array;
+    } else if (targetBuffer.type === 'Uint16Array') {
+      numBytes = length * 2;
+      TypedArray = Uint16Array;
+    } else if (targetBuffer.type === 'Int16Array') {
+      numBytes = length * 2;
+      TypedArray = Uint16Array;
+    } else {
+      throw new Error('TargetBuffer should be Float32Array or Uint8Array');
+    }
+  } else {
+    // Use Uint8 if no targetBuffer is provided
+    numBytes = length;
+    TypedArray = Uint8Array;
+  }
+
+  const imageScalarData = new TypedArray(length);
+
+  const image: IImage = {
+    imageId: options.imageId,
+    referenceImageId: imageId,
+    intercept: 0,
+    windowCenter: 0,
+    windowWidth: 0,
+    color: false,
+    numComps: 1,
+    slope: 1,
+    minPixelValue: 0,
+    maxPixelValue: 255,
+    voiLUTFunction: undefined,
+    rows: imagePlaneModule.rows,
+    columns: imagePlaneModule.columns,
+    sizeInBytes: imageScalarData.byteLength | numBytes,
+    getPixelData: () => imageScalarData,
+    getCanvas: undefined, // todo: which canvas?
+    height: imagePlaneModule.rows,
+    width: imagePlaneModule.columns,
+    rgba: undefined, // todo: how
+    columnPixelSpacing: imagePlaneModule.columnPixelSpacing,
+    rowPixelSpacing: imagePlaneModule.rowPixelSpacing,
+    invert: false,
+  };
+
+  const imageLoadObject = {
+    promise: Promise.resolve(image),
+  };
+  cache.putImageLoadObject(image.imageId, imageLoadObject);
+
+  return imageLoadObject.promise;
+}
+
+/**
+ * Load and cache a list of imageIds
+ *
+ * @param imageIds - list of imageIds
+ * @param options - options for loader
+ *
+ */
+export function createAndCacheDerivedImages(
+  imageIds: Array<string>
+): DerivedImages {
+  if (!imageIds || imageIds.length === 0) {
+    throw new Error(
+      'createAndCacheDerivedImages: parameter imageIds must be list of image Ids'
+    );
+  }
+
+  const derivedImageIds = [];
+  const allPromises = imageIds.map((imageId) => {
+    const options: DerivedImageOptions = {
+      imageId: getDerivedImageId(imageId),
```

Comment:
You don't need to getDerivedImageId here, the `derived` imageId should be part of argument to get passed inside the function, take a look at the createDerivedVolume and see how that has options including volumeId

Reply:
I pass getDerivedImageId as a function param